src/MiddlewareListener.php

```diff
@@ -69,7 +80,13 @@ public function onDispatch(MvcEvent $event)
             foreach ($routeMatch->getParams() as $key => $value) {
                 $psr7Request = $psr7Request->withAttribute($key, $value);
             }
-            $return = $middleware($psr7Request, Psr7Response::fromZend($response));
+            $return = $pipe(
```

Comment:
Yes, this seems reasonable, though I note that `__invoke` does the wrapping in the `CallableDelegateDecorator` for us, but I have no strong feelings either way. If `__invoke` is eventually going to go, then yes the switch makes sense!

`setResponsePrototype` is already called in `createPipeFromSpec` btw, so I don't think needs calling again (unless I missed something)